src/main/java/com/gregtechceu/gtceu/utils/GTUtil.java

```diff
@@ -452,6 +458,46 @@ public static boolean canSeeSunClearly(Level world, BlockPos blockPos) {
         } else return world.isDay();
     }
 
+    /**
+     * @param state the blockstate to check
+     * @return if the block is a snow layer or snow block
+     */
+    public static boolean isBlockSnow(@NotNull BlockState state) {
+        return state.is(Blocks.SNOW) || state.is(Blocks.SNOW_BLOCK);
+    }
+
+    /**
+     * Attempt to break a (single) snow layer at the given BlockPos.
+     * Will also turn snow blocks into snow layers at height 7.
+     *
+     * @return true if the passed IBlockState was valid snow block
+     */
+    public static boolean tryBreakSnow(@NotNull Level level, @NotNull BlockPos pos, @NotNull BlockState state,
+                                       boolean playSound) {
+        boolean success = false;
+        if (state.is(Blocks.SNOW_BLOCK)) {
+            level.setBlock(pos, Blocks.SNOW.defaultBlockState().setValue(SnowLayerBlock.LAYERS, 7),
+                    Block.UPDATE_ALL_IMMEDIATE);
+            success = true;
+        } else if (state.getBlock() == Blocks.SNOW) {
+            int layers = state.getValue(SnowLayerBlock.LAYERS);
+            if (layers == 1) {
+                level.destroyBlock(pos, false);
+            } else {
+                level.setBlock(pos, Blocks.SNOW.defaultBlockState().setValue(SnowLayerBlock.LAYERS, layers - 1),
+                        Block.UPDATE_ALL_IMMEDIATE);
+            }
+            success = true;
+        }
+
+        if (success && playSound) {
+            level.playSound(null, pos.getX(), pos.getY(), pos.getZ(), SoundEvents.LAVA_EXTINGUISH,
```

Comment:
A while back, I figured out that you can actually do this here:
```suggestion
            level.playSound(null, pos, SoundEvents.LAVA_EXTINGUISH,
```